geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/RebalanceManagementDunitTest.java

```diff
@@ -36,92 +37,110 @@
+    int locator1Port = locator1.getPort();
+    MemberVM locator2 =
+        cluster.startLocatorVM(1, l -> l.withHttpService().withConnectionToLocator(locator1Port));
+    cluster.startServerVM(2, "group1", locator1Port);
+    cluster.startServerVM(3, "group2", locator1Port);
+
+    client1 = new ClusterManagementServiceBuilder()
+        .setHost("localhost")
+        .setPort(locator1.getHttpPort())
+        .build();
+    client2 = new ClusterManagementServiceBuilder()
+        .setHost("localhost")
+        .setPort(locator2.getHttpPort())
         .build();
-    gfsh.connect(locator);
 
     // create regions
     Region regionConfig = new Region();
     regionConfig.setName("customers1");
     regionConfig.setType(RegionType.PARTITION);
-    client.create(regionConfig);
-    locator.waitUntilRegionIsReadyOnExactlyThisManyServers("/customers1", 2);
+    client1.create(regionConfig);
+    locator1.waitUntilRegionIsReadyOnExactlyThisManyServers("/customers1", 2);
 
     regionConfig = new Region();
     regionConfig.setName("customers2");
     regionConfig.setType(RegionType.PARTITION);
-    client.create(regionConfig);
-    locator.waitUntilRegionIsReadyOnExactlyThisManyServers("/customers2", 2);
+    client1.create(regionConfig);
+    locator1.waitUntilRegionIsReadyOnExactlyThisManyServers("/customers2", 2);
   }
 
   @Test
-  public void rebalance() throws Exception {
-    ClusterManagementOperationResult<RebalanceResult> cmr =
-        client.start(new RebalanceOperation());
-    assertThat(cmr.isSuccessful()).isTrue();
+  public void rebalance() {
+    ClusterManagementOperationResult<RebalanceOperation, RebalanceResult> startResult =
+        client1.start(new RebalanceOperation());
+    assertThat(startResult.isSuccessful()).isTrue();
     long now = System.currentTimeMillis();
-    assertThat(cmr.getOperationStart().getTime()).isBetween(now - 60000, now);
+    assertThat(startResult.getOperationStart().getTime()).isBetween(now - 60000, now);
 
-    RebalanceResult result = cmr.getFutureResult().get();
-    long end = cmr.getFutureOperationEnded().get().getTime();
+    ClusterManagementOperationResult<RebalanceOperation, RebalanceResult> endResult =
+        waitForStartToEnd(client1, startResult.getOperationId());
+    long end = endResult.getOperationEnd().getTime();
     now = System.currentTimeMillis();
     assertThat(end).isBetween(now - 60000, now)
-        .isGreaterThanOrEqualTo(cmr.getOperationStart().getTime());
+        .isGreaterThanOrEqualTo(endResult.getOperationStart().getTime());
+    RebalanceResult result = endResult.getOperationResult();
     assertThat(result.getRebalanceRegionResults().size()).isEqualTo(2);
     RebalanceRegionResult firstRegionSummary = result.getRebalanceRegionResults().get(0);
     assertThat(firstRegionSummary.getRegionName()).isIn("customers1", "customers2");
   }
 
+  private ClusterManagementOperationResult<RebalanceOperation, RebalanceResult> waitForStartToEnd(
```

Comment:
this is something that I had hoped that could be built in the result that's coming back from the "start" operation.

Reply:
We decided as a team to look into adding a way for the java api user to block until the started op has completed.

Reply:
We added "getFuture" to the ClusterManagementService. We thought this was cleaner than adding "getFuture" to ClusterManagementOperationResult because if we did that it would need to have a reference to the ClusterManagementService to repeatedly call "get".